### DISCUSSION TITLE
can we modify DMV query datatype

### DISCUSSION THREAD

**Discussion by abhinayrozer** [DaxStudio/DaxStudio#909] [Q&A]
Hi folks,
I want to get  single column from DMV query result and i am trying to get that by changing it's type in the query
for example 
select * from $SYSTEM.TMSCHEMA_TABLES

Instead of selecting all columns I can select only desired columns.
like :select [ID],[NAME] from $SYSTEM.TMSCHEMA_TABLES

and here ID is numerical column I want it in text type.

I tried using CAST and convert Functions for ID column. But no luck I am getting below error.

Can please some one tell "is there any way to achieve this ??"
![image](https://user-images.githubusercontent.com/30872272/187151415-f2a25907-7655-4893-bdb6-92191f9544fe.png)

**Comment by marcosqlbi** [comment]
The SQL available for DMV is very limited - I think you cannot cast column data types.

**Reply by abhinayrozer** [reply]
Thanks for the replay Marco
Actually I am trying to get DMV queries executed on Power Bi Dataset using REST API query "executequeries".(using Python).
I am able to get results for non numeric output. But for numeric columns it's throwing error shown in below picture. 
I attached both working and not working queries below.

If you have any thoughts about it please provide.
![MicrosoftTeams-image (15)](https://user-images.githubusercontent.com/30872272/187157738-4deff895-de9d-462d-ad64-327ff54c1aa4.png)
![MicrosoftTeams-image (14)](https://user-images.githubusercontent.com/30872272/187157743-93baa76c-0ad8-46bc-91b7-d3b895f05a66.png)
( 
![MicrosoftTeams-image (13)](https://user-images.githubusercontent.com/30872272/187157502-f1e5466a-8da8-49f6-993b-3ad87d1d0568.png)
![MicrosoftTeams-image (12)](https://user-images.githubusercontent.com/30872272/187157506-8e0d3d2c-3806-4d53-bcf3-0305ee491ab2.png)

**Comment by marcosqlbi** [comment]
I don't see any error.

**Comment by abhinayrozer** [comment]
![MicrosoftTeams-image (13)](https://user-images.githubusercontent.com/30872272/187160083-38e99d4d-56bd-4d62-bfc8-87bca8d759b0.png)

I Attached this images above.

**Reply by dgosbell** [reply]
the ExecuteQueries REST endpoint only supports DAX queries. The DMV queries are using the DMX query language not DAX so what you are trying to do is not supported. You could possibly use something like the scanner api to get metadata details.
